packages/databricks-vscode/src/file-managers/DatabricksEnvFileManager.ts

```diff
@@ -0,0 +1,173 @@
+import {logging} from "@databricks/databricks-sdk";
+import {Loggers} from "../logger";
+import {Context, context} from "@databricks/databricks-sdk/dist/context";
+
+export class DatabricksEnvFileManager implements Disposable {
+    private disposables: Disposable[] = [];
+    private readonly databricksEnvPath: Uri;
+    private readonly userEnvPath: Uri;
+
+    constructor(
+        workspacePath: Uri,
+        private readonly featureManager: FeatureManager,
+        private readonly connectionManager: ConnectionManager,
+        private readonly extensionContext: ExtensionContext
+    ) {
+        const systemVariableResolver = new SystemVariables(workspacePath);
+        const unresolvedDatabricksEnvFile = path.join(
+            "${workspaceFolder}",
+            ".databricks",
+            ".databricks.env"
+        );
+        this.databricksEnvPath = Uri.file(
+            systemVariableResolver.resolve(unresolvedDatabricksEnvFile)
+        );
+
+        const unresolvedUserEnvFile = workspaceConfigs.userEnvFile
+            ? workspaceConfigs.userEnvFile
+            : !workspaceConfigs.msPythonEnvFile ||
+              workspaceConfigs.msPythonEnvFile.includes(
+                  this.databricksEnvPath.fsPath
+              )
+            ? path.join("${workspaceFolder}", ".env")
+            : workspaceConfigs.msPythonEnvFile;
+        this.userEnvPath = Uri.file(
+            systemVariableResolver.resolve(unresolvedUserEnvFile)
+        );
+
+        workspaceConfigs.msPythonEnvFile = unresolvedDatabricksEnvFile;
+        workspaceConfigs.userEnvFile = unresolvedUserEnvFile;
+
+        const userEnvFileWatcher = workspace.createFileSystemWatcher(
+            this.userEnvPath.fsPath
+        );
+
+        this.disposables.push(
+            userEnvFileWatcher,
+            userEnvFileWatcher.onDidChange(() => this.writeFile(), this),
+            userEnvFileWatcher.onDidDelete(() => this.writeFile(), this),
+            userEnvFileWatcher.onDidCreate(() => this.writeFile(), this),
+            featureManager.onDidChangeState(
+                "debugging.dbconnect",
+                () => {
+                    this.writeFile();
+                    this.emitToTerminal();
+                },
+                this
+            ),
+            this.connectionManager.onDidChangeCluster(async (cluster) => {
+                if (!cluster || this.connectionManager.state !== "CONNECTED") {
+                    return;
+                }
+                this.writeFile();
+                this.emitToTerminal();
+            }, this),
+            this.connectionManager.onDidChangeState(async () => {
+                if (this.connectionManager.state !== "CONNECTED") {
+                    return;
+                }
+                this.writeFile();
+                this.emitToTerminal();
+            }, this)
+        );
+    }
+
+    async getPatToken() {
+        const headers: Record<string, string> = {};
```

Comment:
This should use the metadata service so we don't leak PATs to the file system.
I understand that currently, we can't do that yet because DBConnect doesn't use the SDK yet. Let's at least leave a TODO.